pkg/api/service/warnings.go

```diff
@@ -49,13 +50,39 @@ func GetWarningsForService(service, oldService *api.Service) []string {
 		warnings = append(warnings, getWarningsForIP(field.NewPath("spec").Child("loadBalancerIP"), service.Spec.LoadBalancerIP)...)
 	}
 
+	// warning on duplicate ports with different protocols
+	if len(service.Spec.Ports) > 1 {
+		warnings = append(warnings, getDupPortsWarningsForService(service.Spec.Ports)...)
+	}
+
 	for i, cidr := range service.Spec.LoadBalancerSourceRanges {
 		warnings = append(warnings, getWarningsForCIDR(field.NewPath("spec").Child("loadBalancerSourceRanges").Index(i), cidr)...)
 	}
 
 	return warnings
 }
 
+// warning if there are duplicate ports no matter if the protocol is different.
```

Comment:
This is more contentious - can you move it to a different PR?  In this case, what they are doing is VALID and makes sense, but has risk.  We should discuss with sig-net whether to warn (annoying) or not.

Reply:
Removed. I will open a PR later for this part.